src/functions.js

```diff
@@ -36,40 +42,59 @@ function cities( capitals, formatter ){
   // such as 'Paris is the capital of France'
 
   // apply formatter to each object in capitals array and return an array of resulting sentences
+return capitals.map(function(mm){return `${mm.city} is the capital of ${mm.country}`;})
 }
 
 function largerThanTen( numbers ){
   // numbers is an array of numbers
   // return a new array that contains only numbers from the input array which are greater than 10
   // hint: filter
+  return numbers.filter(function(number) { return number > 10;})
 }
 
 function even( numbers ){
   // numbers is an array of numbers
   // return a new array that contains only even numbers from the input array
   // hint: you may want to use the modulus operator '%'
+  return numbers.filter(function(numbe) { return !(numbe % 2);})
 }
 
 function findTheNeedle( words ){
   // words is an array of words
   // return the index of the word 'needle'
   // hint: indexOf
+  return words.indexOf('needle');
 }
 
 function findLargest( numbers ){
   // numbers is an array of numbers
   // return the largest number from that array
+  return Math.max(...numbers)
+  // return numbers.reduce(function(a, b) {return Math.max(a, b)})
 }
 
 function addAllnumbers( numbers ) {
   // numbers is an array of numbers
   // return the sum of all the numbers in the array
+ return numbers.reduce(function(a, b) { return a + b})
 }
 
 function average( things ) {
   // things is an array of numbers and strings
   // return the average of all the numbers
   // be sure to exclude the strings
+  // var result=0,len=0;
+  // for (var i=0;i<things.length; i++){
+  //   if(Number.isInteger(things[i])){
+  //     len++;
+  //     result+=things[i];
+  //   }
+  // }
+  // return result/len;
+  var num =  things.filter(function(item){return typeof item =='number';});
+  var leng = num.length;
+  var sum = num.reduce(function(a, b) {return a + b;});
```

Comment:
How could you make this a bit more concise? `num` is an array, so you can "chain" array methods together:

```js
foo.filter(...).reduce(...)
```